color.go

```diff
@@ -1,101 +1,18 @@
 package lipgloss
 
 import (
-	"sync"
+	"fmt"
 
 	"github.com/muesli/termenv"
 )
 
-var (
-	colorProfile         termenv.Profile
-	getColorProfile      sync.Once
-	explicitColorProfile bool
-
-	hasDarkBackground       bool
-	getBackgroundColor      sync.Once
-	explicitBackgroundColor bool
-
-	colorProfileMtx sync.RWMutex
-)
-
-// ColorProfile returns the detected termenv color profile. It will perform the
-// actual check only once.
-func ColorProfile() termenv.Profile {
-	colorProfileMtx.RLock()
-	defer colorProfileMtx.RUnlock()
-
-	if !explicitColorProfile {
-		getColorProfile.Do(func() {
-			colorProfile = termenv.EnvColorProfile()
-		})
-	}
-	return colorProfile
-}
-
-// SetColorProfile sets the color profile on a package-wide context. This
-// function exists mostly for testing purposes so that you can assure you're
-// testing against a specific profile.
-//
-// Outside of testing you likely won't want to use this function as
-// ColorProfile() will detect and cache the terminal's color capabilities
-// and choose the best available profile.
-//
-// Available color profiles are:
-//
-// termenv.Ascii (no color, 1-bit)
-// termenv.ANSI (16 colors, 4-bit)
-// termenv.ANSI256 (256 colors, 8-bit)
-// termenv.TrueColor (16,777,216 colors, 24-bit)
-//
-// This function is thread-safe.
-func SetColorProfile(p termenv.Profile) {
-	colorProfileMtx.Lock()
-	defer colorProfileMtx.Unlock()
-
-	colorProfile = p
-	explicitColorProfile = true
-}
-
-// HasDarkBackground returns whether or not the terminal has a dark background.
-func HasDarkBackground() bool {
-	colorProfileMtx.RLock()
-	defer colorProfileMtx.RUnlock()
-
-	if !explicitBackgroundColor {
-		getBackgroundColor.Do(func() {
-			hasDarkBackground = termenv.HasDarkBackground()
-		})
-	}
-
-	return hasDarkBackground
-}
-
-// SetHasDarkBackground sets the value of the background color detection on a
-// package-wide context. This function exists mostly for testing purposes so
-// that you can assure you're testing against a specific background color
-// setting.
-//
-// Outside of testing you likely won't want to use this function as
-// HasDarkBackground() will detect and cache the terminal's current background
-// color setting.
-//
-// This function is thread-safe.
-func SetHasDarkBackground(b bool) {
-	colorProfileMtx.Lock()
-	defer colorProfileMtx.Unlock()
-
-	hasDarkBackground = b
-	explicitBackgroundColor = true
-}
-
-// TerminalColor is a color intended to be rendered in the terminal. It
-// satisfies the Go color.Color interface.
+// TerminalColor is a color intended to be rendered in the terminal.
 type TerminalColor interface {
-	value() string
-	color() termenv.Color
-	RGBA() (r, g, b, a uint32)
```

Comment:
Not sure if anybody uses this interface, but since it's public we should strive to keep the `RGBA` function in there. You can probably just copy my implementation from the `instances` branch.

Reply:
We'd have to change the signature of `RGBA()` to `RGBA(*Renderer)` though. Without the renderer, we wouldn't be able to detect dark background colors :/ This affects `AdaptiveColor` and `CompleteColor`.

Reply:
Correct, but instead of breaking the compatibility, I suggest behaving like before and returning the color value for the default output.